dwave/plugins/torch/autoencoder/ae.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2025 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The use of the auto encoder implementations below (including the
+# AutoEncoder) with a quantum computing system is
+# protected by the intellectual property rights of D-Wave Quantum Inc.
+# and its affiliates.
+#
+# The use of the auto encoder implementations below (including the
+# AutoEncoder) with D-Wave's quantum computing
+# system will require access to D-Wave’s LeapTM quantum cloud service and
+# will be governed by the Leap Cloud Subscription Agreement available at:
+# https://cloud.dwavesys.com/leap/legal/cloud_subscription_agreement/
+#
+
+from collections.abc import Callable
+
+import torch
+
+__all__ = ["AutoEncoder"]
+
+
+class AutoEncoder(torch.nn.Module):
+    """Autoencoder architecture amenable for training discrete models as priors.
```

Comment:
in this case, I'd say rename the class to `DiscreteAutoencoder`

footnote: we should be consistent in lower/upper casing of "autoencoder", e.g., AutoEncoder vs Autoencoder

Reply:
I have renamed it to `DiscreteAutoEncoder` and I am also using the name consistently